app/code/community/Oyst/OneClick/controllers/Checkout/CartController.php

```diff
@@ -95,9 +109,66 @@ public function orderAction()
     private function sendResponse()
     {
         $this->getResponse()->setHttpResponseCode(200);
+
         if ('cgi-fcgi' === php_sapi_name()) {
             $this->getResponse()->setHeader('Content-type', 'application/json');
         }
+
         $this->getResponse()->setBody(Zend_Json::encode($this->data));
     }
+
+    /**
+     * Validate Oyst order status
+     *
+     * @param $oystOrderId
+     *
+     * @return bool
+     */
+    private function isOystOrderStatusValid($oystOrderId)
```

Comment:
Cette méthode a du sens d'être positionnée dans un model et non pas dans le controller car elle contient de la logique métier, de plus elle peut être appelée à d'autres endroits y compris en script php console. De plus il faudrait sortir le set du $this->data = Mage::getBaseUrl()...., ce n'est pas la responsabilité d'une méthode GET de faire cela. cette méthode doit juste renvoyer true ou false.

A l'endroit ou elle est appellée on devrait avoir 
if(Mage::getModel()->isOystOrderStatusValid()) {
  $this->data = Mage::getBaseUrl()+success_suffix
} else {
  $this->data = Mage::getBaseUrl()+error_suffix
  return;
}